core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala

```diff
@@ -86,6 +86,16 @@ abstract class FileCommitProtocol {
   def newTaskTempFile(taskContext: TaskAttemptContext, dir: Option[String], ext: String): String
 
   /**
+   * Similar to newTaskTempFile(), but allows files to committed to an absolute output location.
+   * Depending on the implementation, there may be weaker guarantees around adding files this way.
+   */
+  def newTaskTempFileAbsPath(
```

Comment:
Can you unify this and `newTaskTempFile`? If we treat the default partition location like custom location.